components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateUsernameApiServiceImpl.java

```diff
@@ -45,14 +47,17 @@ public Response validateUsernamePost(UsernameValidationRequestDTO user) {
         }
 
         try {
-            String tenantDomain = MultitenantUtils.getTenantDomain(user.getUsername());
             List<PropertyDTO> propertyDTOList = user.getProperties();
+            String tenantDomain = MultitenantUtils.getTenantDomain(user.getUsername());
             boolean skipSelfSignUpEnabledCheck = false;
 
             if (CollectionUtils.isNotEmpty(propertyDTOList)) {
                 for (PropertyDTO propertyDTO : propertyDTOList) {
                     if (SKIP_SIGN_UP_ENABLE_CHECK_KEY.equalsIgnoreCase(propertyDTO.getKey())) {
                         skipSelfSignUpEnabledCheck = Boolean.parseBoolean(propertyDTO.getValue());
+                    } else if (StringUtils.equals(IdentityManagementEndpointConstants.TENANT_DOMAIN,
```

Comment:
we can use IdentityManagementEndpointConstants.TENANT_DOMAIN.equals(propertyDTO.getKey()) here since  IdentityManagementEndpointConstants.TENANT_DOMAIN is a constant